### PR TITLE
[ASButtonNode] Support the combined [.Selected, .Highlighted] state

### PR DESCRIPTION
**Pull request by tomizimobile**

### CODE REVIEW

File: AsyncDisplayKit/ASButtonNode.mm [hunk 1]
```diff
@@ -365,6 +388,30 @@ - (void)setImage:(UIImage *)image forState:(ASControlState)state
...
+        return _selectedHighlightedBackgroundImage;
+      
+      case ASControlStateDisabled:
+        return _disabledBackgroundImage;
+            
+      default:
+        return _normalBackgroundImage;
+    }
+}
+
```
Comment (aaronschubert0) [inline AsyncDisplayKit/ASButtonNode.mm]: @tomizimobile Seems like the spacing for this has reverted to 4 spaces. The standard for ASDK is 2 spaces. You can change this in your project settings. I fell for this a lot initially.

Comment (tomizimobile) [inline AsyncDisplayKit/ASButtonNode.mm]: @aaronschubert0 whoops, the spacing was right but must've gotten blown away after I moved it up to match the order of other getter/setter combos. Fixed.

File: AsyncDisplayKit/ASButtonNode.mm [hunk 2]
```diff
@@ -365,52 +388,58 @@ - (void)setImage:(UIImage *)image forState:(ASControlState)state
...
-- (void)setBackgroundImage:(UIImage *)image forState:(ASControlState)state
+- (UIImage *)backgroundImageForState:(ASControlState)state
 {
   ASDN::MutexLocker l(_propertyLock);
   switch (state) {
     case ASControlStateNormal:
-      _normalBackgroundImage = image;
-      break;
-      
+      return _normalBackgroundImage;
```
Comment (appleguy) [inline AsyncDisplayKit/ASButtonNode.mm]: Thanks!  I definitely prefer the direct return.

**Comment by appleguy** [comment]
@tomizimobile thanks, this is an important feature / subtlety that was not handled in the original!

**Comment by appleguy** [comment]
In the future I would love to have some of this stuff more abstracted so we don't need separate ivars for everything, and the rules can be codified in a more logical way, but this isn't important.

**Pull request merged**